### ISSUE TITLE
Args are not symbolized when passing "symbolize_args: true"

### ISSUE THREAD

**Issue by francktrouillez** [sidekiq-cron/sidekiq-cron#443]
### Steps to reproduce

1. Create a `Sidekiq::Cron::Job` with `args` being `{ my_key: "my_value" }` (using symbols as keys), and `symbolize_args: true`
2. Enqueue the job
3. Observe that the `args` given to the sidekiq worker is `{ "my_key" => "my_value" }` (using strings as keys)

```ruby
# frozen_string_literal: true

require "bundler/inline"

gemfile(true) do
  source "https://rubygems.org"

  gem "sidekiq-cron", git: "https://github.com/sidekiq-cron/sidekiq-cron.git"
end

require "minitest/autorun"
require 'sidekiq/testing'

Sidekiq::Testing.inline! # to perform jobs immediately in tests
Sidekiq.strict_args!(false) # to allow non-native JSON types as job arguments (e.g. symbols)

$worker_args = nil # to store worker arguments

class MyWorker
  include Sidekiq::Worker

  def perform(args)
    $worker_args = args
  end
end

class BugTest < Minitest::Test
  def test_bug
    args = { my_key: "my_value" }
    job = Sidekiq::Cron::Job.new(name: "test", cron: "0 0 * * *", class: "MyWorker", args:, symbolize_args: true)
    job.enque!

    assert_equal(args, $worker_args)
  end
end
```

### Expected behavior

Expect `args` (given to `Sidekiq::Cron::Job` initializer) to be equal to `$worker_args` (given to `MyWorker#perform`). I expect both to be equal to `{ my_key: "my_value" }`

### Actual behavior

`args` and `$worker_args` are different. `args` has the keys symbolized (since it is done in the variable assignment), but `$worker_args` kept the keys as strings.
![image](https://github.com/sidekiq-cron/sidekiq-cron/assets/57403591/127eeb60-ebc4-44c9-a7be-59380b3df19a)

### System configuration
**Ruby version**: 3.2.0

**Comment by markets** [comment]
@francktrouillez would you like to send a patch?

**Comment by francktrouillez** [comment]
@markets sure, I'd be happy to help!

**TL:DR;** [`sidekiq` doesn't allow to pass (and retrieve) args as symbols](https://github.com/sidekiq/sidekiq/wiki/The-Basics#:~:text=These%20two%20methods,not%20serialize%20properly.). That's why I don't think we should allow the `symbolize_args` option in `Sidekiq::Cron::Job`.

I took a look and I have the feeling this is a problem that is going out of the scope of `sidekiq-cron`.

From my understanding, `sidekiq-cron` is a tool to run `MyWorker.perform_async` at a given schedule (so, basically filling the queue the sidekiq worker will consume). In this case, it takes place inside the [`Sidekiq::Cron::Job#enqueue!`](https://github.com/sidekiq-cron/sidekiq-cron/blob/8a04c08ab1819d7f97070c9169bbbe14296c8286/lib/sidekiq/cron/job.rb#L55-L83) that is calling `Sidekiq::Cron::Job#enqueue_sidekiq_worker` that is calling `Sidekiq::Worker#perform_async`.

To me, that's where the scope of `sidekiq-cron` stops.

However, when checking how the symbolization of args works with `sidekiq`, I saw that [sidekiq currently doesn't allow to pass (and retrieve) args as symbols](https://github.com/sidekiq/sidekiq/wiki/The-Basics#:~:text=These%20two%20methods,not%20serialize%20properly.).

<details>
  <summary>I used this test script to test the symbolization of args with sidekiq in ruby 3.2.0</summary>

  ```ruby
  # frozen_string_literal: true

  require "bundler/inline"

  gemfile(true) do
    source "https://rubygems.org"

    gem "sidekiq-cron", git: "https://github.com/sidekiq-cron/sidekiq-cron.git"
  end

  require "minitest/autorun"
  require 'sidekiq/testing'

  Sidekiq::Testing.inline! # to perform jobs immediately in tests
  Sidekiq.strict_args!(false) # to allow non-native JSON types as job arguments (e.g. symbols)

  $worker_args = nil # to store worker arguments

  class MyWorker
    include Sidekiq::Worker

    def perform(args)
      $worker_args = args
    end
  end

  class BugTest < Minitest::Test
    def test_bug
      args = { my_key: "my_value" }
      MyWorker.perform_async(args)

      assert_equal(args, $worker_args)
    end
  end
  ```

  I get the following result:
  ![image](https://github.com/sidekiq-cron/sidekiq-cron/assets/57403591/8d3336aa-cdd9-4618-9d2c-6e47e170ebfe)
  So, the args are not symbolized anymore.
</details>

I also found this [issue](https://github.com/sidekiq/sidekiq/issues/3893) that relates to this problem, but since sidekiq relies on JSON strings stored in Redis, they are currently not willing to allow this symbolization as far as I know.

Related issues:
- https://github.com/sidekiq/sidekiq/issues/5221
- https://github.com/sidekiq/sidekiq/issues/6131
- https://github.com/sidekiq/sidekiq/issues/3893
- ...

In order to enable the `symbolize_args` option in `Sidekiq::Cron::Job`, we would need to change the way `sidekiq` works (and I don't think this is the scope of `sidekiq-cron`).

I think it is doable to do it in the scope of `sidekiq-cron` (monkey-patching the sidekiq worker class dynamically when we instantiate the `Sidekiq::Cron::Job` class to overwrite the `#perform` method ) but I don't think it is a good idea, as it changes the sidekiq worker globally and not only in the context of `sidekiq-cron`.

Based on this, I believe that instead of allowing the `symbolize_args` option in `Sidekiq::Cron::Job`, we should smoothly deprecate it and potentially remove it in the next major version of `sidekiq-cron` as it is not working as expected and can be confusing.

What do you think? Does it make sense to you? Let me know if you have any questions or if I missed something.

Thanks!

**Comment by markets** [comment]
WOW! Thanks! such a detailed explanation @francktrouillez 👏🏼 

In fact, my plan for next release is to bump a major version `v2.0` (we merged "namespaces" https://github.com/sidekiq-cron/sidekiq-cron/commit/160d34df350741944d0ed55b75c1a3f5d6b352c8 and I'd also like to remove official support for some things like: EOL Rubies, old Sidekiq, old Redis, ...). So it sounds like a good timing to take some action. 

However, as per our docs it seems that this only works under Active Job? https://github.com/sidekiq-cron/sidekiq-cron?tab=readme-ov-file#active-job-worker I'm not sure to be honest, I wasn't here yet when this code was introduced (#275). Apparently, we also have some passings tests ✅. I'm a bit confused now 😅.

**Comment by francktrouillez** [comment]
@markets Thanks for your quick answer!

Indeed, I missed that this option was in fact only supported for `ActiveJob` and that it was written in the doc, thanks for pointing that out!

However, I still think that since we specify whether we use `ActiveJob` or not using the `active_job` attribute, and that by default, it uses `sidekiq`, I feel we could have something to notify developers about this behavior like:
- Raise an exception when we use `symbolize_arguments` with `sidekiq` when we initialize the `Sidekiq::Cron::Job`
- Log a warning when we use `symbolize_arguments` with `sidekiq` when we initialize the `Sidekiq::Cron::Job`
- Add a comment in the code to document the use of `symbolize_arguments`
- Specifically add to the documentation that `symbolize_arguments` doesn't work/isn't supported by `sidekiq` in the [README](https://github.com/sidekiq-cron/sidekiq-cron?tab=readme-ov-file#sidekiq-worker).
- Something else?
- Do nothing, as it is indeed written in the doc of `ActiveJob` that `symbolize_args` works with it, implying it doesn't with `sidekiq`

Thanks!

**Comment by markets** [comment]
I agree, we can improve the current situation. The first 4 points are super ok to me 👌, as they improve the feedback for the developers. Feel free to send a PR.

Thanks!

**Comment by francktrouillez** [comment]
@markets I sent a [PR](https://github.com/sidekiq-cron/sidekiq-cron/pull/449) to log a warning, feel free to review it.

I didn't dare raising an exception for now, as it can be breaking for people 'silently' using `symbolize_args: true` with `sidekiq`.

What do you think?

What do you think?
